rasa/core/channels/hangouts.py

```diff
@@ -254,21 +264,22 @@ def _extract_input_channel(self) -> Text:
         return self.name()
 
     def _check_token(self, bot_token: Text) -> None:
-        # see https://developers.google.com/hangouts/chat/how-tos/bots-develop#verifying_bot_authenticity # noqa: E501, W505
+        # see https://developers.google.com/chat/how-tos/bots-develop#verifying_bot_authenticity # noqa: E501, W505
+        # and https://google-auth.readthedocs.io/en/latest/user-guide.html#identity-tokens # noqa: E501, W505
         try:
-            token = client.verify_id_token(
-                bot_token, self.project_id, cert_uri=CERT_URI
+            decoded_token = id_token.verify_token(
```

Comment:
I only had a brief look at the docs and GH code, I see there's also a [`verify_oauth2_token` function](https://github.com/googleapis/google-auth-library-python/blob/e6278a815895e050e57fc516f086b4bc89a0864a/google/oauth2/id_token.py#L143), what's the reason for selecting `verify_token`?

Reply:
`verify_token` is the more general version of the functionality, it can verify service requests such as those coming from Google Chat API. verify_oauth2_token and verify_firebase_token are more specialized functions, and you can't supply certification URLs as they are hardcoded (in earlier version of the code I had mistakenly removed the cert URL). It's a bit confusing :)